src/classes/CON_PrimaryAffls_TDTM.cls

```diff
@@ -69,68 +69,72 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
         //Map of Contact ID to list of Account IDs that represent key Affl fields that have been cleared or modified. 
         //The old Affls need to be made non-primary.
         Map<ID, List<ID>> changedConToAccsMap = new Map<ID, List<ID>>();
-        
-        if(newlist != null) {
+
+        // AFTER INSERT
+        if( newlist != null && triggerAction == TDTM_Runnable.Action.AfterInsert ) {
+
+            // Loop over contact list
             for (integer i=0; i<newlist.size(); i++) {
+
                 Contact newContact = (Contact)newlist[i];
-                
-                if(triggerAction == TDTM_Runnable.Action.AfterInsert && !alreadyRunAfterInsert) {
-                    
-                    //If the Contact has been created with any of the Key Affiliation fields populated
-                    //  --> Create a matching primary Affiliation for the Contact
-                    for(Integer j = 0; j < afflMapper.primaryAfflFieldNames.size(); j++) {
-                        try {
-		                    if(newContact.get(afflMapper.primaryAfflFieldNames[j]) != null) {
-		                        UTIL_Debug.debug('****Contact has been created with one of the Key Affiliation fields populated');
-		                        createAfflFromLookup(newContact, j, afflsToInsert);                           
-		                    }
-                        } catch(Exception e) {
-                            UTIL_Debug.debug('****Exception message: ' + e.getMessage());
-                            UTIL_Debug.debug('****Stack trace: \n' + e.getStackTraceString());
-                            newContact.addError(e.getMessage());
+
+                //If the Contact has been created with any of the Key Affiliation fields populated
+                //  --> Create a matching primary Affiliation for the Contact
+                for(Integer j = 0; j < afflMapper.primaryAfflFieldNames.size(); j++) {
+                    try {
+                        if(newContact.get(afflMapper.primaryAfflFieldNames[j]) != null) {
+                            UTIL_Debug.debug('****Contact has been created with one of the Key Affiliation fields populated');
+                            createAfflFromLookup(newContact, j, afflsToInsert);                           
                         }
+                    } catch(Exception e) {
+                        UTIL_Debug.debug('****Exception message: ' + e.getMessage());
+                        UTIL_Debug.debug('****Stack trace: \n' + e.getStackTraceString());
+                        newContact.addError(e.getMessage());
                     }
-                    alreadyRunAfterInsert = true;
```

Comment:
We still want to have re-entrant flags, just not inside the for loop. That is, we want all records for the transaction to be processed, but the method to process them no more than once in the same transaction.